pkgs/applications/science/electronics/kicost/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, python3Packages, fetchFromGitHub }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "kicost";
+  version = "1.1.6";
+
+  src = fetchFromGitHub {
+    owner = "hildogjr";
+    repo = "KiCost";
+    rev = "v${version}";
+    sha256 = "kFuANwdEtGvPltTeC1vY1eWM7TY/3k4GyzdAG/lZLuo=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    beautifulsoup4
+    lxml
+    XlsxWriter
+    tqdm
+    requests
+    validators
+    wxPython_4_1
+    colorama
+  ];
+
```

Comment:
Please try to run the tests https://github.com/hildogjr/KiCost/tree/master/tests

Reply:
@SuperSandro2000 Sorry, got distracted with other projects… I have updated to the latest release 1.1.15. Tests cannot be run though because test (artifacts) are missing in the Python source distribution (both with KiCost and the new dependency kicost-digikey-api-v3). I will get in touch with the maintainers to get this fixed in future releases.

Reply:
In addition, the KiCost test suite fails when trying to spawn a dummy web server:

```
tests/test_kicost.py:295: in run_test
    server = subprocess.Popen(TESTDIR + '/dummy-web-server.py', stdout=fo, stderr=fe)
/nix/store/j652sgyb3137c19v5vy5ziaarw5k5bf6-python3-3.9.15/lib/python3.9/subprocess.py:951: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
...
E               FileNotFoundError: [Errno 2] No such file or directory: '/build/kicost-1.1.15/tests/dummy-web-server.py'
```

Although test artifacts are missing in the source distribution (which would cause failure at a later stage), `dummy_web_server.py` is present at this location; so I suppose this error is caused by Nix build isolation. Any hints on how to work around this issue?

Reply:
> Although test artifacts are missing in the source distribution (which would cause failure at a later stage), `dummy_web_server.py` is present at this location; so I suppose this error is caused by Nix build isolation. Any hints on how to work around this issue?

looks like the shebang could be hardcoded to an absolute path and needs a call with patchShebang